lib/Sema/TypeCheckError.cpp

```diff
@@ -1627,6 +1627,10 @@ void TypeChecker::checkFunctionErrorHandling(AbstractFunctionDecl *fn) {
   // by the time we got here.
   if (!fn->hasInterfaceType()) return;
 
+  // If we skipped typechecking the body, don't perform this check; it will
+  // always warn.
+  if (fn->wasBodyTypecheckingSkipped()) return;
```

Comment:
Would be better if checkFunctionErrorHandling() didn't get called on these functions at all